tracer/test/Datadog.Trace.Tests/Telemetry/Metrics/MetricTests.cs

```diff
@@ -0,0 +1,286 @@
+﻿// <copyright file="MetricTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.SourceGenerators;
+using Datadog.Trace.Telemetry;
+using Datadog.Trace.Telemetry.Metrics;
+using FluentAssertions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using Xunit;
+
+namespace Datadog.Trace.Tests.Telemetry.Metrics;
+
+public class MetricTests
+{
+    private static readonly Dictionary<string, List<string>> IgnoredTagsByMetricName = new()
+    {
+        { "waf.init", new() { "event_rules_version" } }, // we don't send this tag as cardinality is infinite
+        { "waf.updates", new() { "event_rules_version" } }, // we don't send this tag as cardinality is infinite
+        { "waf.requests", new() { "event_rules_version" } }, // we don't send this tag as cardinality is infinite
+        { "spans_finished", new() { "integration_name" } }, // this is technically difficult for us, so we don't tag it
+    };
+
+    private static readonly Dictionary<string, List<string>> OneOfTagsByMetricName = new()
+    {
+        { "init_time", new() { "total", "component" } }, // we only send one of these
+    };
+
+    [Fact]
+    public void OnlyAllowedMetricsAreSubmitted()
+    {
+        // Only metrics defined in the following json documents should be submitted
+        // https://github.com/DataDog/dd-go/trace/apps/tracer-telemetry-intake/telemetry-metrics/static/common_metrics.json
+        // https://github.com/DataDog/dd-go/trace/apps/tracer-telemetry-intake/telemetry-metrics/static/dotnet_metrics.json
```

Comment:
Should we try to download these files and compare them to the local copy?

Reply:
Yeah, I considered that, but I wasn't sure if that would complicate things because the repo is private...

There's also the workflow problem - you really want to add them to the local copy + tracer _before_ making a PR to the telemetry intake. So you don't want to _always_ use the live version.

I was also wondering about a GitHub action that does this and creates a PR to update them, but that has some of the same issues, so I'm not really sure what's best :/